examples/gitlab/README.md

```diff
@@ -0,0 +1,54 @@
+# What is this?
+
+This is an example [Opta](https://github.com/run-x/opta) configuration file to deploy [Gitlab](https://docs.gitlab.com/charts/) on AWS.
+
+You can also use this button to fill in the variable fields in the config.
+
+[![Deploy](https://raw.githubusercontent.com/run-x/opta/main/assets/deploy-to-aws-button.svg)](https://app.runx.dev/deploy-with-aws?url=https%3A%2F%2Fgithub.com%2Frun-x%2Fopta%2Fblob%2Fmain%2Fexamples%2Fgitlab%2Fgitlab-aws.yaml&name=Gitlab)
+
+# What does this do?
+It deploys a horizontally scalable Gitlab deployment on EKS in AWS. It also sets up various other resources like VPCs, subnets, load balancers etc.
+
+# Steps to deploy
+* Fill in the required variables in the config file, gitlab-aws.yaml.
+* Run `opta apply -c gitlab-aws.yaml` on the config file
+* You will be prompted to approve changes a couple of times as opta will go through various stages. This is expected, just keep approving.
+
+That's it. Gitlab is deployed on AWS. You can find the AWS load balancer URL to access the deployment by running `opta output` (looks for the field called `load_balancer_raw_dns`).
+
+**Note**: In addition to this, you can also attach RDS and Redis instances, as shown [here](https://docs.gitlab.com/charts/charts/globals.html) by properly integrating Secrets with the Kubernetes.
+
+**NOTE**: DNS and SSL/TLS is currently not set up. To get them to work, follow the next section.
+
+# Getting DNS to work
+* Run `opta output -c gitlab-aws.yaml` to get the nameservers. You will see a section like this:
+```text
+name_servers = tolist([
+  “ns-1384.awsdns-45.org”,
+  “ns-2001.awsdns-58.co.uk”,
+  “ns-579.awsdns-08.net”,
+  “ns-83.awsdns-10.com”,
+])
+```
+* Go to your domain registrar (link namecheap, godaddy, etc.) to point the domain to these nameservers.
+* Update `delegated` field to `true` in the `gitlab-aws.yaml` file
+* Run `opta apply -c gitlab-aws.yaml` again to generate the TLS certificate
+
+Your domain should now be pointing to the Ghost deployment with secure TLS
+
+* The admin dashboard would be available at https://gitlab.{your-domain}/
+* To find the password for the root user:
+```shell
+$> echo $(kubectl get secrets -n gitlab gitlab-gitlab-gitlab-initial-root-password -o json | jq '.data.password')
+"OTdjNklub0wzOFlYVWRRSENhRWJhMzRZeHRkNEFmV0NoWjVZcmw0VDlMTXp5eDlBYldrZXBvenJERWxwM2pxcQ=="
+$> echo "OTdjNklub0wzOFlYVWRRSENhRWJhMzRZeHRkNEFmV0NoWjVZcmw0VDlMTXp5eDlBYldrZXBvenJERWxwM2pxcQ==" | base64 -D
+97c6InoL38YXUdQHCaEba34Yxtd4AfWChZ5Yrl4T9LMzyx9AbWkepozrDElp3jqq
```

Comment:
```suggestion
$ kubectl get secrets -n gitlab gitlab-gitlab-gitlab-initial-root-password -o json | jq -r '.data.password' | base64 -D
example_password
```

- `echo $( ... )` is superfluous
- Using "real" secret values, even in examples is generally not a good idea because sometimes people copy those example values and use them for real, and automated scanners can pick them up (and then "report them" to us in an issue)
- Its better to just keep piping the data through instead of making the user copy/paste

Reply:
passing values to pipe was essentially failing, and I didn't want this to become a bad user experience.
As for the actual values, I'll replace them with dummy.

Reply:
What was "essentially failing"? I added the `-r` flag to `jq` (output raw values) which drops the quote marks from the output. Other than that, I can't think of what would cause failures.